packages/mask/src/plugins/NextID/components/tip/TipTaskManager.tsx

```diff
@@ -0,0 +1,37 @@
+import { EMPTY_LIST } from '@masknet/web3-shared-evm'
+import { FC, useCallback, useEffect, useState } from 'react'
+import { TipTaskProvider } from '../../contexts'
+import { PluginNextIdMessages } from '../../messages'
+import type { TipTask } from '../../types'
+import { TipDialog } from './TipDialog'
+
+let id = 0
+
+interface Task extends TipTask {
+    id: number
+}
+export const TipTaskManager: FC = ({ children }) => {
+    const [tasks, setTasks] = useState<Task[]>(EMPTY_LIST)
+
+    const removeTask = useCallback((task: Task) => {
+        setTasks((list) => list.filter((t) => t !== task))
+    }, [])
+
+    useEffect(() => {
+        PluginNextIdMessages.tipTask.on((task) => {
```

Comment:
To tear down the listener.

```suggestion
        return PluginNextIdMessages.tipTask.on((task) => {
```